api/v1alpha1/addresspool_conversion.go

```diff
@@ -13,11 +13,20 @@ func (src *AddressPool) ConvertTo(dstRaw conversion.Hub) error {
 	dst.Spec.Protocol = src.Spec.Protocol
 	dst.Spec.Addresses = make([]string, len(src.Spec.Addresses))
 	copy(dst.Spec.Addresses, src.Spec.Addresses)
-	*dst.Spec.AutoAssign = *src.Spec.AutoAssign
-	dst.Spec.BGPAdvertisements = make([]v1beta1.LegacyBgpAdvertisement, 0)
+	if src.Spec.AutoAssign != nil {
+		autoAssign := *src.Spec.AutoAssign
+		dst.Spec.AutoAssign = &autoAssign
```

Comment:
How about utility copy functions? Another option is to use the function of the pointer package, that will allocate a new pointer based on the value you are passing in.